### PR TITLE
fix breadcrumb API & add get_page_crumbs method

### LINKED ISSUE #1365: rx.breadcrumb should support lists
Currently we need to manually create `rx.breadcrumb_item`s within to fill this out. We should have a high level constructor similar to the other components (e.g. rx.slider)to support initializing with lists.

### PR DESCRIPTION
**Pull request by Lendemor**
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### New Feature Submission:

- [x] Does your submission pass the tests? 
- [x] Have you linted your code locally prior to submission?

### Description

Fix the breadcrumb API so they can be used with Var & ComputedVar.

Also add an utils method `get_page_crumbs()` which give page crumb in a compatible format with `rx.breadcrumb`

closes #1365 and #1366 

### Examples
```
class MyState()
    crumbs: list[tuple[str, str]] = []

    def on_load(self):
        self.crumbs = self.get_page_crumbs()

    @rx.var
    def computed_crumbs(self) -> list[tuple[str, str]]:
        return self.get_page_crumbs()

def index():
    return rx.breadcrumb(items=MyState.crumbs)

def index_2():
    return rx.breadcrumb(items=MyState.computed_crumbs)
```

### CODE REVIEW

File: reflex/components/navigation/breadcrumb.py [hunk 1]
```diff
@@ -31,11 +33,18 @@ def create(cls, *children, items=None, **props) -> Component:
             The breadcrumb component.
         """
         if len(children) == 0:
-            children = []
-            for label, link in items or []:
-                children.append(
-                    BreadcrumbItem.create(BreadcrumbLink.create(label, href=link))
-                )
+            if isinstance(items, (Var, ComputedVar)):
```
Comment (picklelo) [inline reflex/components/navigation/breadcrumb.py:36]: I think checking just `Var` should handle `ComputedVar` as it's a subclass

File: reflex/components/navigation/breadcrumb.py [hunk 2]
```diff
@@ -31,11 +33,18 @@ def create(cls, *children, items=None, **props) -> Component:
...
+                children = [
+                    Foreach.create(
```
Comment (picklelo) [inline reflex/components/navigation/breadcrumb.py:38]: Nice!

File: reflex/state.py [hunk 3]
```diff
@@ -484,6 +484,23 @@ def get_current_page(self) -> str:
         """
         return self.router_data.get(constants.RouteVar.PATH, "")
 
+    def get_page_crumbs(self) -> List[Tuple[str, str]]:
```
Comment (picklelo) [inline reflex/state.py:487]: Maybe this method can be called `get_path_parts` ? Crumbs (without breadcrumbs) is a bit confusing

File: reflex/state.py [hunk 4]
```diff
@@ -484,6 +484,23 @@ def get_current_page(self) -> str:
...
+        """Obtain page crumbs based on current path and query params.
+
+        Returns:
+            List[Tuple[str, str]]: the breacrumbs in a format easy to use with Breadcrumb component.
+        """
+        path = self.get_current_page()
+        for k, v in self.get_query_params().items():
```
Comment (picklelo) [inline reflex/state.py:494]: Can you add some comments on what these parts of the code are doing

**Review by picklelo — COMMENTED** [review]
Looks good, just some minor comments

**Comment by markthepixel** [comment]
thx :)

**Review by picklelo — APPROVED** [review]
awesome, looks good

**Pull request merged**